.github/workflows/build-plugin-archive.yml

```
@@ -0,0 +1,241 @@
+          fi
+
+      - name: Set up custom environment variables
+        env:
+          ENV_VARS: ${{ secrets.ENV_VARS }}
+        if: ${{ env.ENV_VARS }}
+        uses: actions/github-script@v6
+        with:
+          script: |
+            JSON
+              .parse(process.env.ENV_VARS)
+              .forEach(envVar => core.exportVariable(envVar.name, envVar.value));
+
+      #
+      # PHP dependencies
+      #
+
+      - name: Set up PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ inputs.PHP_VERSION }}
+
+      - name: Install Composer dependencies with dev dependencies
+        # ⬑ The idea is to ensure we have non-production tools like composer-assets-compiler and translation downloader available
+        uses: ramsey/composer-install@v2
+        with:
+          composer-options: ${{ inputs.COMPOSER_ARGS_DEV }}
+
+      - name: Check optional composer build tools
+        id: composer-tools
+        run: |
+          hasDep(){
+            composer show -i -N -D --strict "${1}" >/dev/null 2>&1
+            local EXIT=$?
+            echo "$EXIT"
+          }
+          echo "assets-compiler=$( hasDep inpsyde/composer-assets-compiler )" >> $GITHUB_OUTPUT
+          echo "translation-downloader=$( hasDep inpsyde/wp-translation-downloader )" >> $GITHUB_OUTPUT
+
+      - name: Run Composer Assets Compiler
+        if: steps.composer-tools.outputs.assets-compiler == '0'
+        run: composer compile-assets
+
+      - name: Run WP Translation Downloader
+        if: steps.composer-tools.outputs.translation-downloader == '0'
+        run: composer wp-translation-downloader:download
+
+      - name: Install Composer dependencies without dev dependencies
+        # ⬑ Now we want to get rid of dev dependencies and most probably skip composer scripts
+        uses: ramsey/composer-install@v2
+        with:
+          composer-options: ${{ inputs.COMPOSER_ARGS }}
+
+
+      #
+      # JS environment. Only for setting up node/npm versions. Actual tasks should be run at project level
+      #
+
+      - name: Check for package.json
+        id: check-for-package
+        run: |
+          if [ -f package.json ]; then
+            echo "has_package=true" >> $GITHUB_OUTPUT
+          else
+            echo "has_package=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Set up node
+        if: steps.check-for-package.outputs.has_package == 'true'
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ inputs.NODE_VERSION }}
+          registry-url: ${{ inputs.NPM_REGISTRY_DOMAIN }}
+
+      - name: Cache npm dependencies
+        if: steps.check-for-package.outputs.has_package == 'true'
+        uses: actions/cache@v3
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
```

No need for the `actions/cache` package.
```suggestion
          cache: ${{ inputs.PACKAGE_MANAGER }}
```

---

Cool!